scripts/check-esgf-scrape.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+# Check our scrape of ESGF
+#
+# If the scrape has changed,
+# automatically generate a new PR.
+
+# To run this script, you need an environment with GitHub's CLI installed
+# On nimbus, you can get into that with:
+# - `mamba init && source /home/jovyan/.bashrc && mamba activate /shared/mamba-gh/`
+#
+# You will also have needed to either have logged in using GitHub's CLI
+# or set the `GH_TOKEN` variable using a GitHub API authentication token.
+
+AUTO_GENERATED_PATH="/p/user_pub/work/input4MIPs/esgf-input4MIPs.json"
+REPO_PATH="Database/input-data/esgf-input4MIPs.json"
+
+cp $AUTO_GENERATED_PATH $REPO_PATH
+
+git add $REPO_PATH
+git diff-index --cached --quiet HEAD
+
+if [ $? -eq 0 ]; then
+    echo "No changes after grabbing the latest scrape"
+    exit 0
+fi
+
+
+TIMESTAMP=$(date +%Y-%m-%d_%H-%M-%S)
+BRANCH_NAME="${TIMESTAMP}_esgf-input4mips-json-update"
+git checkout -b "${BRANCH_NAME}"
+git commit -m "Update ESGF input4MIPs JSON file"
+git push --set-upstream origin "${BRANCH_NAME}"
+gh pr create --title "Bot: ${TIMESTAMP} update ESGF JSON" --body "Created from nimbus bot" --label "esgf-json-update" --reviewer znichollscr #--reviewer durack1
```

Comment:
is this `gh` an alias of yours?

Reply:
Nah it's an official github tool, see e.g. https://anaconda.org/conda-forge/gh

Reply:
Other installation options here https://github.com/cli/cli#linux--bsd

Reply:
Sheez, yet another dependency... We're starting to drift off the KISS principle considerably..

Reply:
> We're starting to drift off the KISS principle considerably

I mean, yes, with a few extra thoughts.

First: this is totally optional, we can also just change

```sh
git commit -m "Update ESGF input4MIPs JSON file"
git push --set-upstream origin "${BRANCH_NAME}"
gh pr create --title "Bot: ${TIMESTAMP} update ESGF JSON" --body "Created from nimbus bot" --label "esgf-json-update" --reviewer znichollscr --reviewer durack1
```

to

```sh
git commit -m "Update ESGF input4MIPs JSON file @znichollscr @durack1"
git push --set-upstream origin "${BRANCH_NAME}"
```

and we'll still get notified I think, we just have to make the MR by hand.

Second thought: the GitHub CLI is already what Dan is using for making all the automated PRs, issues etc. so this complexity is already in our stack. Rolling our own solution would be much more headache than using what GitHub provides (which seems pretty good and stable to me). Anyway, as I said, totally optional to use the GitHub CLI here.

Reply:
> the GitHub CLI is already what Dan is using for making all the automated PRs, issues etc. so this complexity is already in our stack. 

Fair comment, in order to leverage automated functionality there will be an increment in dependence that has to occur, more dependencies == less manual work (maybe). So point taken.

I'll have to figure out what is needed in an env to make this work, but let's merge with the machine/crontab configuration a second step which requires the merge in place anyway. We don't have an open issue for this do we? So if this is merge we'll have to hunt for it as a remaining to-do

Reply:
Just to make sure I understand, the idea would be that the crontab job would also make the MR? If yes, makes sense. I can help with environment set up.

I'd suggest picking this back up again once we're through the CEDS/volcanic work. It could be a bit fiddly, so best to do without too many other moving parts going at the same time I think.

Reply:
> If you mean Pull Request (PR), exactly

Oops yes my bad.